src/System.Net.Security/src/System/Net/Security/SslStreamInternal.cs

```diff
@@ -167,43 +157,149 @@ internal int EndRead(IAsyncResult asyncResult)
 
         internal IAsyncResult BeginWrite(byte[] buffer, int offset, int count, AsyncCallback asyncCallback, object asyncState)
         {
-            var lazyResult = new LazyAsyncResult(this, asyncState, asyncCallback);
-            ProcessWrite(buffer, offset, count, lazyResult);
-            return lazyResult;
+            return TaskToApm.Begin(WriteAsync(buffer, offset, count, CancellationToken.None), asyncCallback, asyncState);
         }
 
-        internal void EndWrite(IAsyncResult asyncResult)
+        internal void EndWrite(IAsyncResult asyncResult) => TaskToApm.End(asyncResult);
+
+        internal Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
         {
-            if (asyncResult == null)
+            ValidateParameters(buffer, offset, count);
+
+            SslWriteAsync writeAdapter = new SslWriteAsync(_sslState, cancellationToken);
+
```

Comment:
Nit: there's a blank line here but not in the corresponding Write code.

Reply:
changed